scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -32,23 +32,43 @@
   ; suppress otherwise identical states so we don't spam.
   (= old-state new-state))
 
-(defn remove-finalization-if-set-and-delete
-  [api-client expected-state-dict pod]
-  (api/remove-finalization-if-set-and-delete api-client pod)
-  expected-state-dict)
+(defn delete-task
+  "Kill task is the same as deleting a task. I semantically distinguish them. Delete is used for completed tasks that
+  we're done with. Kill is used for possibly running tasks we want to kill so that they fail. Returns a new expected state
+  dict of nil."
+  [api-client pod]
+  (api/kill-task api-client pod)
+  nil) ; Return new expected state dict of nil.
 
 (defn kill-task
+  "Kill task is the same as deleting a task. I semantically distinguish them. Delete is used for completed tasks that
+  we're done with. Kill is used for possibly running tasks we want to kill so that they fail."
   [api-client expected-state-dict pod]
   (api/kill-task api-client pod)
   expected-state-dict)
 
+(defn weird
+  "Weird. This pod is in a weird state. Log its weird state."
+  [expected-state-dict pod]
+  (log/error "Pod in a weird state:" expected-state-dict "and pod" pod))
+
+(defn kill-task-weird
```

Comment:
Maybe rename to `kill-pod-in-weird-state`

Reply:
pods don't escape the cook.kubernetes.* namespaces. Maybe the original sin was not recognizing 'task' as mesos langauge, and using instance (cook language) here.

Reply:
Afaik, task and instance are different. Task is to refer to mesos entities, instance is to refer to the cook side of things corresponding to the task. In the k8s world, the mesos task is a k8s pod.